addons/account/tests/test_account_journal.py

```diff
@@ -294,3 +294,27 @@ def test_alias_from_type(self):
                          'Journal alias owned by journal itself')
         self.assertEqual(journal_alias_2.alias_parent_thread_id, journal.id,
                          'Journal alias owned by journal itself')
+
+    def test_alias_journal_after_rename(self):
+        """ Test alias when journal is created with name matching existing alias """
+        company_name = 'company_1_data'
```

Comment:
Thank you for the thorough review, it seems I misunderstood some things while reading and testing how the aliases are generated and I will start working on a better version.

However I'm unable to recreate the case where the code is False since the journal cannot be created without a code, in my case the alias is successfully created with the code. What am I missing ?

Reply:
Here is an example video where the code is False: https://app.screencastify.com/v3/watch/bymZIwT2wVgmPntjqBxV